### DISCUSSION TITLE
actual library support please

### DISCUSSION THREAD

**Discussion by BaddRadish** [godotengine/godot-proposals#5483] [Scripting]
autoload is great when you want to load a node, but i'd like to load resource gdscripts. can we get an autoload that has a little bit smarter handling for different types of variables? like not just nodes but resources too? just seems silly to /have to/ attach a script to a node to use it globally. i know you can load a script at the top of every script that uses it, but we have autoload nodes for a reason. -_-

**Comment by Calinou** [comment]
You can add a ResourcePreloader node as an autoloaded scene :slightly_smiling_face:

**Comment by BaddRadish** [comment]
okay so we do have a solution for this in godot already, its just not clear. maybe a tutorial on this would be a good idea, since libraries are a very common concept. maybe there is one, but i havent seen it.

```
class_name Foo
static func bar():
```
then
`Foo.bar()`
